platform-core-module/src/main/java/com/softicar/platform/core/module/file/stored/server/AGStoredFileServer.java

```diff
@@ -13,6 +15,20 @@ public IDisplayString toDisplayWithoutId() {
 		return IDisplayString.create(getUrl());
 	}
 
+	public static List<AGStoredFileServer> getAllActiveWithPrimaryFirst() {
```

Comment:
added validator to prevent the primary server from being deactivated

Reply:
I think this can be circumvented by first deactivating a file server, and then adding it as primary. Solving this via validators appears quite indirect to me. I think we should instead change `getPrimary` to return empty if the primary file server is disabled. Less code, consistent behavior.